### ISSUE TITLE
Termwiz parses F6+ and modifier-F1-4 incorrectly

### ISSUE THREAD

**Issue by jonaskoelker** [wez/wezterm#178]
When I run the program at https://gist.github.com/jonaskoelker/855d611b504c967d700a1ed8972c3fce and press F6, I get the output

```
read terminal settings Termios { inner: RefCell { value: termios { c_iflag: 17664, c_oflag: 5, c_cflag: 191, c_lflag: 35387, c_line: 0, c_cc: [3, 28, 127, 21, 4, 0, 1, 0, 17, 0, 26, 0, 18, 15, 23, 22, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], c_ispeed: 15, c_ospeed: 15 } }, input_flags: ICRNL | IXON | IUTF8, output_flags: OPOST | ONLCR, control_flags: CS6 | CS7 | CS8 | CREAD | CSIZE, local_flags: ECHOKE | ECHOE | ECHOK | ECHO | ECHOCTL | ISIG | ICANON | IEXTEN, control_chars: [3, 28, 127, 21, 4, 0, 1, 0, 17, 0, 26, 0, 18, 15, 23, 22, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] }
set terminal settings: Ok(())
read 5 bytes: [27, 91, 49, 55, 126]
parsed some bytes into Key(KeyEvent { key: Function(7), modifiers: NONE })
```

That is, termwiz parses the F6 byte sequence as F7.

A quick `ascii d27 d91 d49 d55 d126` confirms that the byte sequence is "ESC [ 1 7 ~", the expected encoding of F6. See http://aperiodic.net/phil/archives/Geekery/term-function-keys.html for a list of function key encodings.

Function keys F6 through F10 are all off by one. F11 and F12 give me

```
read 5 bytes: [27, 91, 50, 51, 126]
parsed some bytes into Key(KeyEvent { key: Escape, modifiers: NONE })
parsed some bytes into Key(KeyEvent { key: Char('['), modifiers: NONE })
parsed some bytes into Key(KeyEvent { key: Char('2'), modifiers: NONE })
parsed some bytes into Key(KeyEvent { key: Char('3'), modifiers: NONE })
parsed some bytes into Key(KeyEvent { key: Char('~'), modifiers: NONE })
read 5 bytes: [27, 91, 50, 52, 126]
parsed some bytes into Key(KeyEvent { key: Escape, modifiers: NONE })
parsed some bytes into Key(KeyEvent { key: Char('['), modifiers: NONE })
parsed some bytes into Key(KeyEvent { key: Char('2'), modifiers: NONE })
parsed some bytes into Key(KeyEvent { key: Char('4'), modifiers: NONE })
parsed some bytes into Key(KeyEvent { key: Char('~'), modifiers: NONE })
```

Pressing Shift-F1 gives me
```
read 6 bytes: [27, 91, 49, 59, 50, 80]
parsed some bytes into Key(KeyEvent { key: Escape, modifiers: NONE })
parsed some bytes into Key(KeyEvent { key: Char('['), modifiers: NONE })
parsed some bytes into Key(KeyEvent { key: Char('1'), modifiers: NONE })
parsed some bytes into Key(KeyEvent { key: Char(';'), modifiers: NONE })
parsed some bytes into Key(KeyEvent { key: Char('2'), modifiers: NONE })
parsed some bytes into Key(KeyEvent { key: Char('P'), modifiers: NONE })
```

This behavior is seen in both xterm and tmux-in-xterm.

In tmux-in-xterm, pressing Alt-F1 gives
```
read 4 bytes: [27, 27, 79, 80]
parsed some bytes into Key(KeyEvent { key: Function(1), modifiers: ALT })
```

whereas in xterm I get
```
read 6 bytes: [27, 91, 49, 59, 51, 80]
parsed some bytes into Key(KeyEvent { key: Escape, modifiers: NONE })
parsed some bytes into Key(KeyEvent { key: Char('['), modifiers: NONE })
parsed some bytes into Key(KeyEvent { key: Char('1'), modifiers: NONE })
parsed some bytes into Key(KeyEvent { key: Char(';'), modifiers: NONE })
parsed some bytes into Key(KeyEvent { key: Char('3'), modifiers: NONE })
parsed some bytes into Key(KeyEvent { key: Char('P'), modifiers: NONE })
```

This is on ubuntu 18.04, xterm version 330-1ubuntu2 and tmux 2.6-3ubuntu0.2. Running `xterm -version` gives `XTerm(330)` and `tmux -V` gives `tmux 2.6`.

I'd happily submit a patch. If you'd like that, please let me know. If there's anything I should be aware of before coding up a fix, please let me know about that too :)

**Comment by wez** [comment]
Ah, so much for "robust input processing" :-p

I'll gladly accept a PR!
The relevant section of code is here: 
https://github.com/wez/wezterm/blob/master/termwiz/src/input.rs#L636-L665

The general strategy is that we have some built-in defaults and that we can be augmented with information from terminfo, although it looks like I didn't actually write the latter part of that yet(!)

You can also use `cargo run --example key_tester` to run the input processing end-to-end; its code is here:
https://github.com/wez/wezterm/blob/master/termwiz/examples/key_tester.rs

**Comment by wez** [comment]
This is fixed in termwiz 0.9.0
